prov/tcp/src/tcpx_fabric.c

```
@@ -0,0 +1,319 @@
+		fd_entry = tcpx_fabric->fd_list.next;
+		fd_info_entry = (container_of(fd_entry,
+					struct poll_fd_info,
+					entry));
+		if (*nfds < 1024) {
+			for (i =0 ; i < *nfds ; i++) {
+				if (poll_fds[i].fd == fd_info_entry->sock_fd) {
+					if ((fd_info_entry->flags & TCPX_SOCK_DEL) &&
+					    (FI_CLASS_PEP == fd_info_entry->fid->fclass)) {
+						fd_list_delete_item(i,poll_fds, fd_info, nfds);
+						tcpx_pep = container_of(fd_info_entry->fid,
+									struct tcpx_pep,
+									util_pep.pep_fid.fid);
+						tcpx_pep->cm.sock_fd_closed = 1;
+					}
+					dlist_remove(&tcpx_fabric->fd_list);
+					free(fd_info_entry);
+					continue;
+				}
+			}
+			fd_info[*nfds] = *fd_info_entry;
+			poll_fds[*nfds].fd = fd_info[*nfds].sock_fd;
+			switch (fd_info[*nfds].fid->fclass) {
+			case FI_CLASS_EP:
+				poll_fds[*nfds].events = POLLOUT;
+				(*nfds)++;
+				break;
+			case FI_CLASS_PEP:
+				poll_fds[*nfds].events = POLLIN;
+				(*nfds)++;
+				break;
+			default:
+				FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
+					"invalid fd\n");
+			}
+			dlist_remove(&tcpx_fabric->fd_list);
+			free(fd_info_entry);
+		}
+	}
+	fastlock_release(&tcpx_fabric->fd_list_lock);
+}
+
+static void handle_fd_events(struct tcpx_fabric *fabric,
+			   struct pollfd *poll_fds,
+			   struct poll_fd_info *fd_info,
+			   int *nfds)
+{
+	int iter, status, ret;
+	socklen_t len;
+	struct tcpx_conn_handle *handle;
+	struct util_ep *util_ep;
+	struct tcpx_pep *tcpx_pep;
+	struct fi_eq_cm_entry eq_entry;
+
+	for (iter = 0; iter < *nfds ; iter++) {
+		switch (fd_info[iter].fid->fclass) {
+		case FI_CLASS_EP:
+			util_ep = container_of(fd_info[iter].fid,
+					       struct util_ep,
+					       ep_fid.fid);
+
+			if (poll_fds[iter].revents & POLLOUT) {
+				len = sizeof(status);
+				ret = getsockopt(poll_fds[iter].fd, SOL_SOCKET,
+						 SO_ERROR, &status, &len);
+				if (ret < 0)
+					FI_WARN(&tcpx_prov, FI_LOG_EP_CTRL,
+						"getsockopt failed\n");
+				if (status == 0) {
```

status can be used uninitialized in this function.  I would drop this check.  If getsockopt fails or status is an error, we should report a connection failure to the EQ.